hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystem.java

```diff
@@ -717,7 +800,7 @@ public boolean mkdirs(final Path f, final FsPermission permission) throws IOExce
           listener);
       abfsStore.createDirectory(qualifiedPath,
           permission == null ? FsPermission.getDirDefault() : permission,
-          FsPermission.getUMask(getConf()), tracingContext);
+          FsPermission.getUMask(getConf()), Trilean.UNKNOWN, tracingContext);
```

Comment:
Why do we need a override flag for create directory??
If there is a file present at the path, we are supposed to fail. If there is an implicit directory, we create a marker blob, if there is an explicit directory, then operation should be idempotent.

Am I missing anything here?

Reply:
There is a new arg added in abfsStore.createDirectory(), if the val of arg is UNKNOWN, it will take what the namespace config is given (this happens on trunk, no change in this). But, if there is boolean val in this arg, then that is taken.


This is done because, in the case of nonHNS, it will always overwrite, there is a requriement in the patch, where this is method is required, but doesnt want overwrite to happen.

Reply:
Have added javadocs in the method with definition of param `isOverwriteRequired` for code readablity

Reply:
Have removed it. It is no more required. Thanks!